packages/opentelemetry-sdk-trace-base/src/export/WriteableSpan.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as api from '@opentelemetry/api';
+import { SpanAttributeValue } from '@opentelemetry/api';
+import {
+  InstrumentationLibrary
+} from '@opentelemetry/core';
+import { IResource } from '@opentelemetry/resources';
+import { TimedEvent } from '../TimedEvent';
+
+export interface WriteableSpan extends api.Span {
+    readonly kind: api.SpanKind;
+    readonly parentSpanId?: string;
+    readonly attributes: api.SpanAttributes;
+    readonly links: api.Link[];
+    readonly events: TimedEvent[];
+    readonly startTime: api.HrTime;
+    readonly resource: IResource;
+    readonly instrumentationLibrary: InstrumentationLibrary;
+    name: string;
+    status: api.SpanStatus;
+    endTime: api.HrTime;
+
+    spanContext(): api.SpanContext
+
+    setAttribute(key: string, value?: SpanAttributeValue): this;
+
+    setAttributes(attributes: api.SpanAttributes): this
+
+    addEvent(
+      name: string,
+      attributesOrStartTime?: api.SpanAttributes | api.TimeInput,
+      startTime?: api.TimeInput
+    ): this
+
+    setStatus(status: api.SpanStatus): this
+
+    updateName(name: string): this
+
+    end(endTime?: api.TimeInput): void
+
+    isRecording(): boolean
+
+    recordException(exception: api.Exception, time?: api.TimeInput): void;
```

Comment:
These declarations are all taken from [api `Span` interface](https://github.com/open-telemetry/opentelemetry-js-api/blob/bafcf0d0756e6ecc79fcc75f221b9dd17641eba5/src/trace/span.ts#L32). 
I am not sure but I think they can be omitted if the current interface `extends api.Span`, right? That would also guarantee compatibility of the interface in case of future fixes and changes (not needing to repeat them here as well).